cpp/oneapi/dal/algo/logistic_regression/backend/optimizer_impl.hpp

```diff
@@ -36,7 +36,14 @@ class optimizer_impl : public base {
     virtual double get_tol() = 0;
     virtual std::int64_t get_max_iter() = 0;
 
+    // this function returns meaningful value only for newton_cg optimizer
+    // inner iterations value can be accessed after minimize method was called
+    virtual std::int64_t get_inner_iter() {
+        return -1;
```

Comment:
What about throwing an error instead?

Reply:
I don't think that it is necessary at this place, from my point returning -1 value is sufficient here